turtle_tf2_py/turtle_tf2_py/turtle_tf2_message_broadcaster.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from geometry_msgs.msg import PointStamped
+from geometry_msgs.msg import Twist
+
+import rclpy
+from rclpy.node import Node
+
+from turtlesim.msg import Pose
+from turtlesim.srv import Spawn
+
+
+class PointPublisher(Node):
+
+    def __init__(self):
+        super().__init__('turtle_tf2_message_broadcaster')
+
+        self.client = self.create_client(Spawn, 'spawn')
+        while not self.client.wait_for_service(timeout_sec=1.0):
+            self.get_logger().info('service not available, waiting again...')
+        request = Spawn.Request()
+        request.name = 'turtle3'
+        request.x = float(4)
+        request.y = float(2)
+        request.theta = float(0)
+        self.client.call_async(request)
```

Comment:
I think you have not seen that, as the fixing PR was made not long ago, but we have changed the way request is called in nodes. The issue was is that if the service is not available during that process, the node will got stuck in the `__init__` phase and will not start spinning. What I suggest is to take a look at the update spawn syntax that we came up with and try to modify it in a similar manner:

https://github.com/ros/geometry_tutorials/blob/ros2/turtle_tf2_cpp/src/turtle_tf2_listener.cpp

Reply:
Ok, I will try to modify it in a similar manner.

Reply:
@kurshakuz So I must spawn turtle3 in callback function? Then there will be problem: 
self.sub = self.create_subscription(Pose, 'turtle3/pose', self.handle_turtle_pose, 1)
My callback will be called in the  subscription, and require a positional argument 'msg', and when I launch this node, it will never spawn turtle3. How to resolve this problem?

Reply:
@kurshakuz All other suggestion you made were resolved except for this.

Reply:
I see your point. Do you think we then can make a separate timer call responsible for spawn service? Then we can make a flag that would indicate that the turtle is spawned and could start the subscription process?

Reply:
@kurshakuz I have done it according to your good point, and now it will work.